packages/lambda-tiler/src/routes/fonts.ts

```diff
@@ -0,0 +1,50 @@
+import { Env } from '@basemaps/shared';
+import { fsa } from '@chunkd/fs';
+import { LambdaHttpRequest, LambdaHttpResponse } from '@linzjs/lambda';
+import path from 'path';
+
+interface FontGet {
+  Params: { fontStack: string; range: string };
+}
+
+export async function fontGet(req: LambdaHttpRequest<FontGet>): Promise<LambdaHttpResponse> {
+  const assetLocation = Env.get(Env.AssetLocation);
+  if (assetLocation == null) return new LambdaHttpResponse(404, 'No Found');
+
+  try {
+    const filePath = fsa.join(assetLocation, path.join('fonts', req.params.fontStack, req.params.range)) + '.pbf';
+    const buf = await fsa.read(filePath);
+    const res = new LambdaHttpResponse(200, 'ok');
+    res.buffer(buf, 'application/x-protobuf');
+    return res;
+  } catch (e: any) {
+    if (e.code === 404) return new LambdaHttpResponse(404, 'No Found');
+    throw e;
+  }
+}
+
+export async function fontList(): Promise<LambdaHttpResponse> {
+  const assetLocation = Env.get(Env.AssetLocation);
+  if (assetLocation == null) return new LambdaHttpResponse(404, 'No Found');
+
+  try {
+    const filePath = fsa.join(assetLocation, '/fonts');
+    const fonts = new Set<string>();
+
+    // TODO use {recursive: false}
+    for await (const fontPath of fsa.list(filePath)) {
+      if (!fontPath.endsWith('.pbf')) continue;
+      const basePath = fontPath.slice(filePath.length + 1);
+      const dirName = path.dirname(basePath); // TODO this only works for /a/b.pbf and not /a/b/c.pbf
+      if (dirName.includes('/')) continue;
```

Comment:
Logic here is quite confusing. Are we getting all the fonts directory name without `/`? like `s3://linz-basemaps/font/NewTimes`?

Reply:
Yeah this is annoying as its a recursive list. the next `@chunkd` would make this logic redundant.

basically uou get a list looking like

```
s3://linz-basemaps-static/fonts/Roboto Thin/0-255.pbf
s3://linz-basemaps-static/fonts/Roboto Black/0-255.pbf
```

and you want the output to be ```["Roboto Thin", "Roboto Black"]```

Reply:
I need to work on some unit tests for this, I am making a `FsMemory` to write tests against. so it should make more sense once we have tests!